app/user/page.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { useState, useEffect } from "react";
+
+import Link from "next/link";
+import { useAccount, useReadContract } from "wagmi";
+
+import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+
+import { abi as HatsAbi } from "@/lib/hatsAbi";
+
+export default function AccountPage() {
+  const { address, isConnected } = useAccount();
+  const [hatId, setHatId] = useState<string | null>(null);
+
+  const hatsContractAddress = "0x3bc1A0Ad72417f2d411118085256fC53CBdDd137"; // Hats protocol contract address
+  const hatsId = "0x0000027000020000000000000000000000000000000000000000000000000000"; // https://app.hatsprotocol.xyz/trees/11155111/624?hatId=624.2
+
+  const {
+    data: hatData,
+    isError,
+    isLoading,
+  } = useReadContract({
+    address: hatsContractAddress,
+    abi: HatsAbi,
+    functionName: "viewHat",
+    args: [hatsId],
+    enabled: isConnected,
+  });
+
+  useEffect(() => {
+    if (hatData) {
+      setHatId(hatData.toString());
+    }
+  }, [hatData]);
+
+  if (!isConnected) {
```

Comment:
I'm thinking we should probably create a component out of this. We will need it in many places.

Reply:
Let's do it when we need it